src/lakefs_spec/spec.py

```diff
@@ -646,6 +688,10 @@ def rm_file(self, path: str | os.PathLike[str]) -> None:
             self.client.objects_api.delete_object(
                 repository=repository, branch=branch, path=resource
             )
+            # Directory listing cache for the containing folder must be invalidated
+            parent_path = str(Path(path).parent)
+            if parent_path in self.dircache:
+                del self.dircache[parent_path]
```

Comment:
Also, this is `O(cachesize)` because of the `in` check, you could also `self.dircache.pop(parent_path, None)` for a (hopefully?) `O(1)` solution (not that it matters anywhere outside of lakeFS'ing in Google-sized monorepos)

Reply:
Actually, `DirCache` implements `__contains__` as an element access in a `dict`, which has [average case complexity in `O(1)`](https://wiki.python.org/moin/TimeComplexity). However, as above, `pop` seems like the way to go here.